include/seqan3/io/sequence_file/format_sam.hpp

```diff
@@ -0,0 +1,347 @@
+              // seq_qual field implies that they are the same variable
+              assert(std::addressof(sequence) == std::addressof(qualities));
+              ranges::copy(stream_view | view::char_to<typename value_type_t<qual_type>::quality_alphabet_type>,
+                           ranges::begin(qualities) + sequence_size_before);
+          }
+          else if constexpr (!detail::decays_to_ignore_v<qual_type>)
+          {
+            ranges::copy(stream_view | view::take_until_or_throw(is_cntrl||is_tab)
+                                     | view::char_to<value_type_t<qual_type>>,
+                        ranges::back_insert_iterator{qualities});
+            // In case of tags
+            detail::consume(stream_view | ranges::view::take_while(is_tab));
+            detail::consume(stream_view | ranges::view::take_while(!is_cntrl));
+          }
+          else
+          {
+              detail::consume(stream_view | ranges::view::take_while(!is_cntrl));
+          }
+
+          // make sure "buffer at end" implies "stream at end"
+          if ((std::istreambuf_iterator<char>{stream} == std::istreambuf_iterator<char>{}) &&
+              (!stream.eof()))
+          {
+              stream.get(); // triggers error in stream and sets eof
+          }
+
+          if ((std::istreambuf_iterator<char>{stream} == std::istreambuf_iterator<char>{}) &&
+              (!stream.eof()))
+          {
+              stream.get(); // triggers error in stream and sets eof
+          }
+
+          // Jump over the new line character in the end
+          detail::consume(stream_view | ranges::view::take_while(is_cntrl));
+    }
+
+    //!\copydoc sequence_file_output_format_concept::write
+    template <typename stream_type,     // constraints checked by file
+              typename seq_type,        // other constraints checked inside function
+              typename id_type,
+              typename qual_type>
+    void write(stream_type                     & stream,
+               sequence_file_output_options const & options,
+               seq_type                       && sequence,
+               id_type                        && id,
+               qual_type                      && qualities)
+    {
+        if constexpr (!(detail::decays_to_ignore_v<seq_type>))
+        {
+            static_assert(std::ranges::ForwardRange<seq_type> && alphabet_concept<value_type_t<seq_type>>,
+          "The sequence must model std::ranges::ForwardRange and its value type must model seqan3::alphabet_concept. ");
+        }
+        ranges::ostreambuf_iterator stream_it{stream};
+        // ID
+        if constexpr (detail::decays_to_ignore_v<id_type>)
+        {
+            throw std::logic_error{"The ID field may not be set to ignore when writing sam files."};
+        }
+        else
+        {
+            if (ranges::empty(id)) //[[unlikely]]
+                throw std::runtime_error{"The ID field may not be empty when writing sam files."};
```

Comment:
Actually, everything can be empty in the SAM format, so you can remove the exception and write out `*`